walrus/methods/connectors/azure_connector.py

```diff
@@ -0,0 +1,410 @@
+        """
+        Upload a file to Diffgram from an Azure Blob
+
+        :param opts: Dictionary with parameters for object fetching.
+        :return: file obj if file was uploaded, else False
+        """
+        print('OBJECTCT', opts)
+        spec_list = [{'bucket_name': str, 'path': str}]
+        log = regular_log.default()
+        log, input = regular_input.input_check_many(untrusted_input = opts,
+                                                    spec_list = spec_list,
+                                                    log = log)
+        if len(log["error"].keys()) >= 1:
+            return {'log': log}
+        shared_access_signature = BlobSharedAccessSignature(
+            account_name = self.connection_client.account_name,
+            account_key = self.connection_client.credential.account_key
+        )
+
+        expiration_offset = 40368000
+        blob_name = opts['path']
+        container = opts['bucket_name']
+        added_seconds = datetime.timedelta(0, expiration_offset)
+        expiry_time = datetime.datetime.utcnow() + added_seconds
+        filename = blob_name.split("/")[-1]
+        sas = shared_access_signature.generate_blob(
+            container_name = container,
+            blob_name = blob_name,
+            start = datetime.datetime.utcnow(),
+            expiry = expiry_time,
+            permission = BlobSasPermissions(read = True),
+            content_disposition = 'attachment; filename=' + filename,
+        )
+        sas_url = 'https://{}.blob.core.windows.net/{}/{}?{}'.format(
+            self.connection_client.account_name,
+            container,
+            blob_name,
+            sas
+        )
+
+        with sessionMaker.session_scope() as session:
+
+            project = Project.get_by_string_id(session, self.config_data.get('project_string_id'))
+            member = session.query(Member).filter(Member.user_id == opts['event_data']['request_user']).first()
+            # Deduct Media Type:
+            extension = Path(opts['path']).suffix
+            extension = extension.lower()
+            media_type = None
+            if extension in images_allowed_file_names:
+                media_type = 'image'
+            elif extension in videos_allowed_file_names:
+                media_type = 'video'
+            else:
+                # TODO: Decide, do we want to raise an exception? or just do nothing?
+                log = regular_log.default()
+                log['error']['invalid_type'] = 'File must type of: {} {}'.format(str(images_allowed_file_names),
+                                                                                 str(videos_allowed_file_names))
+                log['error']['file_name'] = opts['path']
+                log['opts'] = opts
+                Event.new(
+                    session = session,
+                    member_id = opts['event_data']['request_user'],
+                    kind = 'aws_s3_new_import_warning',
```

Comment:
@PJEstrada Should we make this more generic or update it to be azure specific or?

Reply:
Yes you're right! Thanks for catching that. I've fixed it here: https://github.com/diffgram/diffgram/pull/35/commits/65dc4e53d981078b6d766bb4de16de3ec208d9cd